kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluator.java

```diff
@@ -299,6 +299,42 @@ ExpressionTransformResult visitLike(final Predicate like) {
       return new ExpressionTransformResult(transformedExpression, BooleanType.BOOLEAN);
     }
 
+    @Override
+    ExpressionTransformResult visitStartsWith(final Predicate startsWith) {
+      if (startsWith.getChildren().size() != 2) {
+        throw unsupportedExpressionException(
+            startsWith,
+            "Invalid number of inputs to STARTS_WITH expression. "
+                + "Example usage: STARTS_WITH(column, 'test')");
+      }
+      ExpressionTransformResult leftResult = visit(startsWith.getChildren().get(0));
+      ExpressionTransformResult rightResult = visit(startsWith.getChildren().get(1));
+      if (!(StringType.STRING.equivalent(leftResult.outputType)
+          && StringType.STRING.equivalent(rightResult.outputType))) {
+        throw unsupportedExpressionException(
+            startsWith, "'starts with' is only supported for string type expressions");
+      }
+      // TODO: support non literal as the second input of starts with.
+      if (!(rightResult.expression instanceof Literal)) {
+        throw unsupportedExpressionException(
+            startsWith, "'starts with' expects literal as the second input");
+      }
+      Literal right = (Literal) rightResult.expression;
+      // Transforms START_WITH(a, "b") to LIKE(a, "b%") if b non-null, escaping all '%' in b.
+      return new ExpressionTransformResult(
+          new Predicate(
+              "LIKE",
+              Arrays.asList(
+                  leftResult.expression,
+                  right.getValue() == null
```

Comment:
can this be null?

Reply:
Technically we could define a null literal with string types
https://github.com/delta-io/delta/blob/master/kernel/kernel-api/src/main/java/io/delta/kernel/expressions/Literal.java#L190-L195

Functional wise
> SELECT startswith('SparkSQL', NULL);
 NULL
[1]https://learn.microsoft.com/en-us/azure/databricks/sql/language-manual/functions/startswith#examples